InnerEye/ML/configs/histo_configs/classification/DeepSMILEPanda.py

```diff
@@ -0,0 +1,143 @@
+import os
+from monai.transforms import Compose
+from pytorch_lightning.callbacks.model_checkpoint import ModelCheckpoint
+from InnerEye.Common import fixed_paths
+from InnerEye.ML.Histopathology.datamodules.panda_module import PandaTilesDataModule
+from InnerEye.Azure.azure_util import get_default_azure_config_json_path
+from InnerEye.ML.Histopathology.datasets.panda_tiles_dataset import PandaTilesDataset
+from health_ml.networks.layers.attention_layers import GatedAttentionLayer
+from InnerEye.ML.Histopathology.models.transforms import (
+    EncodeTilesBatchd,
+    LoadTilesBatchd,
+)
+from InnerEye.ML.Histopathology.models.encoders import (
+    HistoSSLEncoder,
+    ImageNetEncoder,
+    ImageNetSimCLREncoder,
+    InnerEyeSSLEncoder,
+)
+from InnerEye.ML.configs.histo_configs.classification.BaseMIL import BaseMIL
+from InnerEye.ML.configs.histo_configs.run_ids import innereye_ssl_checkpoint
+from health_azure.utils import CheckpointDownloader
+
+
+class DeepSMILEPanda(BaseMIL):
+    def __init__(self, **kwargs: Any) -> None:
+        default_kwargs = dict(
+            # declared in BaseMIL:
+            pooling_type=GatedAttentionLayer.__name__,
+            # declared in DatasetParams:
+            local_dataset=Path("/tmp/datasets/PANDA_tiles"),
+            azure_dataset_id="PANDA_tiles",
+            # To mount the dataset instead of downloading in AML, pass --use_dataset_mount in the CLI
+            # declared in TrainerParams:
+            num_epochs=200,
+            recovery_checkpoint_save_interval=10,
+            recovery_checkpoints_save_last_k=-1,
+            # declared in WorkflowParams:
+            number_of_cross_validation_splits=5,
+            cross_validation_split_index=0,
+            # declared in OptimizerParams:
+            l_rate=5e-4,
+            weight_decay=1e-4,
+            adam_betas=(0.9, 0.99),
+        )
+        default_kwargs.update(kwargs)
+        super().__init__(**default_kwargs)
+        super().__init__(**default_kwargs)
+        self.best_checkpoint_filename = "checkpoint_max_val_auroc"
+        self.best_checkpoint_filename_with_suffix = (
+            self.best_checkpoint_filename + ".ckpt"
+        )
+        self.checkpoint_folder_path = "outputs/checkpoints/"
+        best_checkpoint_callback = ModelCheckpoint(
+            dirpath=self.checkpoint_folder_path,
+            monitor="val/accuracy",
+            filename=self.best_checkpoint_filename,
+            auto_insert_metric_name=False,
+            mode="max",
+        )
+        self.callbacks = best_checkpoint_callback
+
+    @property
+    def cache_dir(self) -> Path:
+        return Path(
+            f"/tmp/innereye_cache1/{self.__class__.__name__}-{self.encoder_type}/"
+        )
+
+    def setup(self) -> None:
+        if self.encoder_type == InnerEyeSSLEncoder.__name__:
+            self.downloader = CheckpointDownloader(
+                azure_config_json_path=get_default_azure_config_json_path(),
```

Comment:
health_azure.utils has a function "get_workspace" that will search the cwd's parents until it finds a config.json - it might be preferable to call that here and get rid of this function, then we are a bit robust to where the user puts the config.json file?

Reply:
Thanks for the suggestion, I'll look for that function and replace

Reply:
done